src/app/components/shared/date-time-filter/filter.component.ts

```diff
@@ -0,0 +1,199 @@
+import { AfterViewInit, Component, EventEmitter, Input, Output, ViewChild } from "@angular/core";
+import { NgForm } from "@angular/forms";
+import { Comparisons, DateExpressions, Filters, InnerFilter } from "@baw-api/baw-api.service";
+import { isInstantiated } from "@helpers/isInstantiated/isInstantiated";
+import { withUnsubscribe } from "@helpers/unsubscribe/unsubscribe";
+import { Project } from "@models/Project";
+import { Region } from "@models/Region";
+import { Site } from "@models/Site";
+import { NgbCalendar, NgbDate, NgbDateParserFormatter } from "@ng-bootstrap/ng-bootstrap";
+import { debounceTime, distinctUntilChanged, takeUntil } from "rxjs";
+import { defaultDebounceTime } from "src/app/app.helper";
+
+export interface Model {
+  projects?: Project[];
+  regions?: Region[];
+  sites?: Site[];
+  dateFiltering?: boolean;
+  dateStartedAfter?: NgbDate;
+  dateFinishedBefore?: NgbDate;
+  todFiltering?: boolean;
+  todIgnoreDst?: boolean;
+  todStartedAfter?: string;
+  todFinishedBefore?: string;
+}
+
+@Component({
+  selector: "baw-date-time-filter",
+  templateUrl: "filter.component.html",
+  styleUrls: ["filter.component.scss"]
+})
+export class DateTimeFilterComponent extends withUnsubscribe() implements AfterViewInit {
+  public constructor(
+    private formatter: NgbDateParserFormatter,
+    private calendar: NgbCalendar
+  ) { super(); }
+
+  @ViewChild(NgForm) public form: NgForm;
+
+  @Input() public project: Project;
+  @Input() public region: Region;
+  @Input() public site: Site;
+  @Output() public constructedFilters = new EventEmitter<Filters>();
+
+  private previousFilters: string;
+  public model: Model = { todIgnoreDst: true };
+  public errors: {
+    todBoundaryError?: boolean;
+  } = {};
+
+  public ngAfterViewInit(): void {
+    this.form.valueChanges
+      .pipe(
+        debounceTime(defaultDebounceTime),
+        distinctUntilChanged(),
+        takeUntil(this.unsubscribe)
+      )
+      .subscribe((model: Model): void => this.updateHref(model));
+  }
+
+  public updateHref(model: Model): void {
+    // validate that the dates are a valid format & are within bounds before sending an API filter request
+    // has the value of null if there is no update to the filters
+    const newFilters = this.generateFilters(model);
+
+    // if the new filters will be the same as the old filters, do not emit a filter update event
+    if (newFilters && this.model) {
```

Comment:
why are we testing if this.model is truthy?

Reply:
This seems to be an artifact from an old version of my code that looked something like this:

![image](https://user-images.githubusercontent.com/33742269/227430655-fcb1362b-51f6-4ed7-8864-3660d131bfcc.png)


I have since removed it